fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/RecordConstructorValue.java

```diff
@@ -161,7 +161,7 @@ private Object deepCopyIfNeeded(@Nonnull TypeRepository typeRepository,
             return null;
         }
 
-        if (fieldType.isPrimitive()) {
+        if (fieldType.isPrimitive() || fieldType instanceof Type.Enum) {
```

Comment:
Is this sufficient? Doesn't the `Descriptors.EnumValueDescriptor.getType` need to match to target field descriptor in the same way as nested messages?